### PR TITLE
feat(runner): Add option to disable auto update

### PR DESCRIPTION
**Pull request by npalm**
Add an option to disable the auto update of the runner. Default behavior is updating the runner to remain backwards compatiblity and keep aligned with the defaults of the runner.

- [x] Add option to the runner lambda
- [x] Add tests to the runner lambda
- [x] Add option to pool
- [x] Add option to Terraform
- [x] Update docs
- [x] Test

### CODE REVIEW

**Comment by npalm** [comment]
Close #1694

**Comment by npalm** [comment]
Close #1694

File: main.tf [hunk 1]
```diff
@@ -107,6 +107,7 @@ module "runners" {
   github_app_parameters                = local.github_app_parameters
   enable_organization_runners          = var.enable_organization_runners
   enable_ephemeral_runners             = var.enable_ephemeral_runners
+  disable_update_runners               = var.disable_update_runners
```
Comment (ScottGuymer) [inline main.tf:110]: Just on naming

Maybe `disable_runner_autoupdate` might be a little easier to grok

**Comment by npalm** [comment]
Tested 

Screenshot 2022-02-25 at 11 26 01

File: modules/runners/scale-up.tf [hunk 2]
```diff
@@ -15,9 +15,15 @@ resource "aws_lambda_function" "scale_up" {
 
   environment {
     variables = {
+      DISABLE_UPDATE_RUNNERS               = var.disable_update_runners
```
Comment (npalm) [inline modules/runners/scale-up.tf:18]: Sorted the list of variables

**Review by ScottGuymer — DISMISSED** [review]
Just one minor comment other than that LGTM

**Comment by npalm** [comment]
@ScottGuymer back to draft, missed the pool. The pool also needs te be configurable.

**Review by ScottGuymer — APPROVED** [review]
:shipit:

**Pull request merged**